tests/sentinel/tests/06-ckquorum.tcl

```diff
@@ -24,9 +38,15 @@ test "CKQUORUM detects failover authorization cannot be reached" {
         kill_instance sentinel [expr {$i + 1}]
     }
 
-    after 5000
-    catch {[S 0 SENTINEL CKQUORUM mymaster]} err
-    assert_match "*NOQUORUM*" $err
+    # Wait for the current sentinel to monitor that other sentinels are
+    # `down`, so that we can check the quorum directly later.
+    wait_for_condition 1000 50 {
+        [get_sentinel_sentinels_down_count 0 mymaster] == $orig_quorum
+    } else {
+        fail "At least $orig_quorum sentinels did not enter the down state."
+    }
+
+    assert_error "*NOQUORUM*" {S 0 SENTINEL CKQUORUM mymaster}
```

Comment:
Maybe, instead of adding logic `get_sentinel_sentinels_down_count`, we can simply `wait_for_condition` on `S 0 SENTINEL CKQUORUM mymaster` to be equal `*NOQUORUM*`.

Reply:
yes, i have the same idea before, like:
```
proc sentinel_chquorum_result {sentinel_id master_name} {
    catch {[S 0 SENTINEL CKQUORUM mymaster]} result
    return $result
}

wait_for_condition 300 50 {
    [string match "*NOQUORUM*" [sentinel_chquorum_result 0 mymaster]]
} else {
    fail "fail"
}
```

but later i changed it , because i also want to verify that this part of the result is expected:
```
    for {set i 0} {$i < $orig_quorum} {incr i} {
        kill_instance sentinel [expr {$i + 1}]
    }
```

but now i suppose in this test case, the simple way is better to make the code understandable.